runtime/interpreter/value.go

```diff
@@ -1785,10 +1782,6 @@ func (v *ArrayValue) Destroy(interpreter *Interpreter, locationRange LocationRan
 
 	config := interpreter.SharedState.Config
 
-	if config.InvalidatedResourceValidationEnabled {
-		v.checkInvalidatedResourceUse(interpreter, locationRange)
-	}
```

Comment:
Like mentioned above, since we do the check after evaluating each expression, these are now redundant and can be removed